src/mapnode.cpp

```diff
@@ -259,142 +259,174 @@ void MapNode::rotateAlongYAxis(const NodeDefManager *nodemgr, Rotation rot)
+				box.MaxEdge.rotateXYBy(180);
+			}
+			else if(facedir == 3)
+			{
 				box.MinEdge.rotateXYBy(-90);
 				box.MaxEdge.rotateXYBy(-90);
-				if(facedir == 1)
-				{
-					box.MinEdge.rotateYZBy(90);
-					box.MaxEdge.rotateYZBy(90);
-				}
-				else if(facedir == 2)
-				{
-					box.MinEdge.rotateYZBy(180);
-					box.MaxEdge.rotateYZBy(180);
-				}
-				else if(facedir == 3)
-				{
-					box.MinEdge.rotateYZBy(-90);
-					box.MaxEdge.rotateYZBy(-90);
-				}
-				break;
-			case 4:  //x-
+			}
+			break;
+		case 2: //z-
+			box.MinEdge.rotateYZBy(-90);
+			box.MaxEdge.rotateYZBy(-90);
+			if(facedir == 1)
+			{
+				box.MinEdge.rotateXYBy(-90);
+				box.MaxEdge.rotateXYBy(-90);
+			}
+			else if(facedir == 2)
+			{
+				box.MinEdge.rotateXYBy(180);
+				box.MaxEdge.rotateXYBy(180);
+			}
+			else if(facedir == 3)
+			{
 				box.MinEdge.rotateXYBy(90);
 				box.MaxEdge.rotateXYBy(90);
-				if(facedir == 1)
-				{
-					box.MinEdge.rotateYZBy(-90);
-					box.MaxEdge.rotateYZBy(-90);
-				}
-				else if(facedir == 2)
-				{
-					box.MinEdge.rotateYZBy(180);
-					box.MaxEdge.rotateYZBy(180);
-				}
-				else if(facedir == 3)
-				{
-					box.MinEdge.rotateYZBy(90);
-					box.MaxEdge.rotateYZBy(90);
-				}
-				break;
-			case 5:
-				box.MinEdge.rotateXYBy(-180);
-				box.MaxEdge.rotateXYBy(-180);
-				if(facedir == 1)
-				{
-					box.MinEdge.rotateXZBy(90);
-					box.MaxEdge.rotateXZBy(90);
-				}
-				else if(facedir == 2)
-				{
-					box.MinEdge.rotateXZBy(180);
-					box.MaxEdge.rotateXZBy(180);
-				}
-				else if(facedir == 3)
-				{
-					box.MinEdge.rotateXZBy(-90);
-					box.MaxEdge.rotateXZBy(-90);
-				}
-				break;
-			default:
-				break;
 			}
-			box.repair();
-			boxes.push_back(box);
+			break;
+		case 3:  //x+
+			box.MinEdge.rotateXYBy(-90);
+			box.MaxEdge.rotateXYBy(-90);
+			if(facedir == 1)
+			{
+				box.MinEdge.rotateYZBy(90);
+				box.MaxEdge.rotateYZBy(90);
+			}
+			else if(facedir == 2)
+			{
+				box.MinEdge.rotateYZBy(180);
+				box.MaxEdge.rotateYZBy(180);
+			}
+			else if(facedir == 3)
+			{
+				box.MinEdge.rotateYZBy(-90);
+				box.MaxEdge.rotateYZBy(-90);
+			}
+			break;
+		case 4:  //x-
+			box.MinEdge.rotateXYBy(90);
+			box.MaxEdge.rotateXYBy(90);
+			if(facedir == 1)
+			{
+				box.MinEdge.rotateYZBy(-90);
+				box.MaxEdge.rotateYZBy(-90);
+			}
+			else if(facedir == 2)
+			{
+				box.MinEdge.rotateYZBy(180);
+				box.MaxEdge.rotateYZBy(180);
+			}
+			else if(facedir == 3)
+			{
+				box.MinEdge.rotateYZBy(90);
+				box.MaxEdge.rotateYZBy(90);
+			}
+			break;
+		case 5:
+			box.MinEdge.rotateXYBy(-180);
+			box.MaxEdge.rotateXYBy(-180);
+			if(facedir == 1)
+			{
+				box.MinEdge.rotateXZBy(90);
+				box.MaxEdge.rotateXZBy(90);
+			}
+			else if(facedir == 2)
+			{
+				box.MinEdge.rotateXZBy(180);
+				box.MaxEdge.rotateXZBy(180);
+			}
+			else if(facedir == 3)
+			{
+				box.MinEdge.rotateXZBy(-90);
+				box.MaxEdge.rotateXZBy(-90);
+			}
+			break;
+		default:
+			break;
+		}
```

Comment:
This code could be much shorter btw. if you rotate by facedir first and then by axisdir.
See: 
https://github.com/minetest/minetest/blob/0530ec11c0371bbe10a22c1800e65ad10234c0b8/src/client/mesh.cpp#L298-L314